server/src/main/java/org/opensearch/telemetry/tracing/listener/TraceableSearchRequestOperationsListener.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.listener;
+
+import org.opensearch.action.search.SearchPhaseContext;
+import org.opensearch.action.search.SearchRequestContext;
+import org.opensearch.action.search.SearchRequestOperationsListener;
+import org.opensearch.telemetry.tracing.AttributeNames;
+import org.opensearch.telemetry.tracing.ScopedSpan;
+import org.opensearch.telemetry.tracing.Span;
+import org.opensearch.telemetry.tracing.SpanBuilder;
+import org.opensearch.telemetry.tracing.SpanContext;
+import org.opensearch.telemetry.tracing.Tracer;
+
+import static org.opensearch.core.common.Strings.capitalize;
+
+/**
+ * SearchRequestOperationsListener subscriber for search request tracing
+ *
+ * @opensearch.internal
+ */
+public final class TraceableSearchRequestOperationsListener extends SearchRequestOperationsListener {
+    private final Tracer tracer;
+    private final Span requestSpan;
+    private ScopedSpan phaseScopedSpan;
+
+    public TraceableSearchRequestOperationsListener(final Tracer tracer, final Span requestSpan) {
+        this.tracer = tracer;
+        this.requestSpan = requestSpan;
+        this.phaseScopedSpan = null;
+    }
+
+    @Override
+    protected void onPhaseStart(SearchPhaseContext context) {
+        assert phaseScopedSpan == null : "There should be only one search phase active at a time";
+        phaseScopedSpan = tracer.startScopedSpan(
```

Comment:
Is it certain that the phaseScopedSpan will be closed on the same thread always? SpanScope is defined to scope the span in the same thread and once the span work is over it should be cleaned from this thread's context. Now here if phaseScopedSpan will be closed in some other thread then the state from this thread won't be cleared.
We should use ScopedSpan only in scenarios where Span will be closed within the same method. Otherwise it is advised to create the Span scope it and propagate the span to track further processing. It's clearly defined in the method javadoc. Please let me know in case amy confusion.

Reply:
This is good point (again this came from me to @dzane17 ), but we indeed have to make sure the thread does not switch (or if it does - use different propagation technique).

Reply:
For coordinator search phases, they can start and end on different threads. Good catch @Gaganjuneja. I will switch it back to use `Span` and `SpanScope` to handle the switching.

Reply:
Like @reta suggested If one phase starts on Thread1 closes on Thread2 then we need to think about the propagation strategy current strategy of holding the SpanScope won't work.